quill-async/src/main/scala/io/getquill/context/async/AsyncContext.scala

```diff
@@ -56,7 +56,7 @@ abstract class AsyncContext[D <: SqlIdiom, N <: NamingStrategy, C <: Connection]
       f(TransactionalExecutionContext(ec, c))
     }
 
-  def executeQuery[T](sql: String, prepare: List[Any] => (List[Any], List[Any]) = row => (Nil, row), extractor: RowData => T = identity[RowData] _)(implicit ec: ExecutionContext): Future[List[T]] = {
+  def executeQuery[T](sql: String, prepare: Prepare = identityPrepare, extractor: Extractor[T] = identityExtractor)(implicit ec: ExecutionContext): Future[List[T]] = {
```

Comment:
I mean changing `identityExtractor` to `identityExtractor _`, compilation fails with:
```
[error] /app/quill-async/src/main/scala/io/getquill/context/async/AsyncContext.scala:70: type mismatch;
[error]  found   : () => AsyncContext.this.ResultRow => AsyncContext.this.ResultRow
[error]     (which expands to)  () => com.github.mauricio.async.db.RowData => com.github.mauricio.async.db.RowData
[error]  required: AsyncContext.this.Extractor[?]
[error]     (which expands to)  com.github.mauricio.async.db.RowData => ?
[error]   def executeQuerySingle[T](sql: String, prepare: Prepare = identityPrepare, extractor: Extractor[T] = identityExtractor _)(implicit ec: ExecutionContext): Future[T] =
[error]
```